### DISCUSSION TITLE
Doppler as Secrets Manager

### DISCUSSION THREAD

**Discussion by kranti07k** [apache/airflow#31097] [Q&A] [answered]
Hello All,

I am planning to use [doppler](https://www.doppler.com/) as secrets management tool for apache airflow. I looked at the documentation https://airflow.apache.org/docs/apache-airflow/1.10.12/howto/use-alternative-secrets-backend.html?highlight=secrets%20manager, how ever I didn't find any resources on using doppler.

Can any one let me know is doppler is supported by apache airflow?

Thanks,
Kranti

**Comment by potiuk** [comment, marked as answer]
Your lins are from 3 years EOL version - you should rather look at https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/security/secrets/secrets-backend/index.html 

and while Doppler is not natively supported you can easily [Roll your own secret backend](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/security/secrets/secrets-backend/index.html#roll-your-own-secrets-backend) - look at existing secret backends and implement your dopler integration. That's expected that you write custom python extensions if they are not supported out-of-the-box.
